### PR TITLE
[ci skip] Add documentation for after_create_commit and after_update_commit callbacks

### PR DESCRIPTION
**Pull request by jagdeepsingh**
This adds the documentation for https://github.com/rails/rails/issues/29554.

### CODE REVIEW

File: guides/source/active_record_callbacks.md [hunk 1]
```diff
@@ -199,6 +199,53 @@ An Employee was touched
...
+    puts 'User was saved to database'
+  end
+end
+
+>> User.create
+=> User was saved to database
+```
+
+WARNING. Using `after_create_commit` and `after_update_commit` both in a model will override the callbacks registered by the first one of them.
+
```
Comment (prathamesh-sonpatki) [inline guides/source/active_record_callbacks.md]: Can you change this to 

> Using `after_create_commit` and `after_update_commit` both in a model will override the callback which was registered first amongst them.

File: guides/source/active_record_callbacks.md [hunk 2]
```diff
@@ -199,6 +199,53 @@ An Employee was touched
 => true
 ```
 
+### `after_create_commit` and `after_update_commit`
```
Comment (prathamesh-sonpatki) [inline guides/source/active_record_callbacks.md]: Please move this at the end of `Transaction Callbacks section`

File: guides/source/active_record_callbacks.md [hunk 3]
```diff
@@ -199,6 +199,53 @@ An Employee was touched
...
+
+The `after_create_commit` and `after_update_commit` macros can be used to register `after_commit` callbacks for Create and Update actions respectively.
```
Comment (prathamesh-sonpatki) [inline guides/source/active_record_callbacks.md]: This line is not needed once we move the section in the end.

File: guides/source/active_record_callbacks.md [hunk 4]
```diff
@@ -199,6 +199,53 @@ An Employee was touched
...
+
+# prints nothing
+>> @user = User.create
+
+# updating @user
+>> @user.save
+=> User was saved to database
+```
+
+To register callbacks for both Create and Update actions, use `after_commit` instead.
```
Comment (prathamesh-sonpatki) [inline guides/source/active_record_callbacks.md]: Please downcase Create and Update.

**Review by prathamesh-sonpatki — CHANGES_REQUESTED** [review]

**Comment by JagdeepSingh** [comment]
@prathamesh-sonpatki thanks for the review. Please check the changes.

File: guides/source/active_record_callbacks.md [hunk 5]
```diff
@@ -416,6 +416,20 @@ common, there are aliases for those operations:
 * `after_destroy_commit`
 
 ```ruby
+class User < ApplicationRecord
```
Comment (prathamesh-sonpatki) [inline guides/source/active_record_callbacks.md]: We don't need it here right?

File: guides/source/active_record_callbacks.md [hunk 6]
```diff
@@ -428,3 +442,32 @@ end
 ```
 
 WARNING. The `after_commit` and `after_rollback` callbacks are called for all models created, updated, or destroyed within a transaction block. However, if an exception is raised within one of these callbacks, the exception will bubble up and any remaining `after_commit` or `after_rollback` methods will _not_ be executed. As such, if your callback code could raise an exception, you'll need to rescue it and handle it within the callback in order to allow other callbacks to run.
+
+WARNING. Using `after_create_commit` and `after_update_commit` both in a model will override the callback which was registered first amongst them.
```
Comment (prathamesh-sonpatki) [inline guides/source/active_record_callbacks.md]: small nitpick, but we can change `a model` to `in the same model`

**Comment by prathamesh-sonpatki** [comment]
@JagdeepSingh Looks good, only few comments and please squash the commits as well into one. Please check http://blog.steveklabnik.com/posts/2012-11-08-how-to-squash-commits-in-a-github-pull-request.

File: guides/source/active_record_callbacks.md (hunk 5 above)
Comment (JagdeepSingh) [inline guides/source/active_record_callbacks.md]: Right. I will remove it.

**Comment by JagdeepSingh** [comment]
Done the changes.

**Review by prathamesh-sonpatki — APPROVED** [review]

**Comment by prathamesh-sonpatki** [comment]
Thanks @JagdeepSingh ❇️

**Pull request merged**